provider/aws.go

```diff
@@ -124,7 +124,9 @@ type AWSProvider struct {
 	zoneTypeFilter ZoneTypeFilter
 	// filter hosted zones by tags
 	zoneTagFilter ZoneTagFilter
-	preferCNAME   bool
+	// only allow changes to specified subdomain and its subdomains
+	subdomainFilter DomainFilter
```

Comment:
should this not be changed to SubdomainFilter instead of DomainFilter?

Reply:
The subdomain filter is still a domain filter and the domain filtering provided by DomainFilter will be sufficient. However, I can add a SubdomainFilter type for readability and future changes. What do you think?

Reply:
Yes please, just thought it would be better to divide it because of readability and future changes as well.